modules/curl/7.69.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,748 @@
++        "lib/security.c",
++        "lib/select.c",
++        "lib/select.h",
++        "lib/sendf.c",
++        "lib/sendf.h",
++        "lib/setopt.c",
++        "lib/setopt.h",
++        "lib/setup-os400.h",
++        "lib/setup-vms.h",
++        "lib/sha256.c",
++        "lib/share.c",
++        "lib/share.h",
++        "lib/sigpipe.h",
++        "lib/slist.c",
++        "lib/slist.h",
++        "lib/smb.h",
++        "lib/smtp.h",
++        "lib/sockaddr.h",
++        "lib/socketpair.h",
++        "lib/socketpair.c",
++        "lib/socks.c",
++        "lib/socks.h",
++        "lib/speedcheck.c",
++        "lib/speedcheck.h",
++        "lib/splay.c",
++        "lib/splay.h",
++        "lib/strcase.c",
++        "lib/strcase.h",
++        "lib/strdup.c",
++        "lib/strdup.h",
++        "lib/strerror.c",
++        "lib/strerror.h",
++        "lib/strtok.c",
++        "lib/strtok.h",
++        "lib/strtoofft.c",
++        "lib/strtoofft.h",
++        "lib/system_win32.h",
++        "lib/telnet.h",
++        "lib/tftp.h",
++        "lib/timeval.c",
++        "lib/timeval.h",
++        "lib/transfer.c",
++        "lib/transfer.h",
++        "lib/url.c",
++        "lib/url.h",
++        "lib/urldata.h",
++        "lib/vauth/cleartext.c",
++        "lib/vauth/cram.c",
++        "lib/vauth/digest.c",
++        "lib/vauth/digest.h",
++        "lib/vauth/ntlm.h",
++        "lib/vauth/oauth2.c",
++        "lib/vauth/vauth.c",
++        "lib/vauth/vauth.h",
++        "lib/version.c",
++        "lib/vssh/ssh.h",
++        "lib/vtls/bearssl.h",
++        "lib/vtls/gskit.h",
++        "lib/vtls/gtls.h",
++        "lib/vtls/mbedtls.h",
++        "lib/vtls/nssg.h",
++        "lib/vtls/openssl.h",
++        "lib/vtls/schannel.h",
++        "lib/vtls/vtls.c",
++        "lib/vtls/vtls.h",
++        "lib/vtls/wolfssl.h",
++        "lib/warnless.c",
++        "lib/warnless.h",
++        "lib/wildcard.c",
++        "lib/wildcard.h",
++        "lib/x509asn1.h",
++        "lib/psl.h",
++        "lib/psl.c",
++        "lib/vtls/sectransp.h",
++        "lib/vtls/mesalink.h",
++        "lib/vtls/mesalink.c",
++        "lib/curl_get_line.h",
++        "lib/curl_get_line.c",
++        "lib/urlapi-int.h",
++        "lib/urlapi.c",
++        "lib/altsvc.h",
++        "lib/altsvc.c",
++        "lib/doh.h",
++        "lib/doh.c",
++    ] + select({
++        ":macos": [
++            "lib/vtls/sectransp.c",
++        ],
++        ":windows": CURL_WIN_SRCS,
++        "//conditions:default": [
++            "lib/vtls/openssl.c",
++        ],
++    }),
++    hdrs = [
++        "include/curl/curl.h",
++        "include/curl/curlver.h",
++        "include/curl/easy.h",
++        "include/curl/mprintf.h",
++        "include/curl/multi.h",
++        "include/curl/stdcheaders.h",
++        "include/curl/system.h",
++        "include/curl/typecheck-gcc.h",
++        "include/curl/urlapi.h",
++    ],
++    copts = select({
++        ":windows": CURL_WIN_COPTS,
```

Comment:
This select on the Windows platform should probably be a select on the MSVC compiler instead (see `@rules_cc//cc/compiler`)